examples/governance/apps/src/bin/publisher.rs

```diff
@@ -0,0 +1,175 @@
+use hex::decode;
+use methods::FINALIZE_VOTES_ELF;
+use risc0_ethereum_contracts::groth16;
+use risc0_zkvm::{default_prover, ExecutorEnv, ProverOpts, VerifierContext};
+
+sol! {
+    interface RiscZeroGovernor {
+        function verifyAndFinalizeVotes(bytes calldata seal, bytes calldata journal) public;
+    }
+}
+
+/// Wrapper of a `SignerMiddleware` client to send transactions to the given
+/// contract's `Address`.
+pub struct TxSender {
+    chain_id: u64,
+    client: SignerMiddleware<Provider<Http>, Wallet<k256::ecdsa::SigningKey>>,
+    contract: Address,
+}
+
+impl TxSender {
+    /// Creates a new `TxSender`.
+    pub fn new(chain_id: u64, rpc_url: &str, private_key: &str, contract: &str) -> Result<Self> {
+        let provider = Provider::<Http>::try_from(rpc_url)?;
+        let wallet: LocalWallet = private_key.parse::<LocalWallet>()?.with_chain_id(chain_id);
+        let client = SignerMiddleware::new(provider.clone(), wallet.clone());
+        let contract = contract.parse::<Address>()?;
+
+        Ok(TxSender {
+            chain_id,
+            client,
+            contract,
+        })
+    }
+
+    /// Send a transaction with the given calldata.
+    pub async fn send(&self, calldata: Vec<u8>) -> Result<Option<TransactionReceipt>> {
+        let tx = TransactionRequest::new()
+            .chain_id(self.chain_id)
+            .to(self.contract)
+            .from(self.client.address())
+            .data(calldata);
+
+        log::info!("Transaction request: {:?}", &tx);
+
+        let tx = self.client.send_transaction(tx, None).await?.await?;
+
+        log::info!("Transaction receipt: {:?}", &tx);
+
+        Ok(tx)
+    }
+}
+
+/// Arguments of the publisher CLI.
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct Args {
+    /// Ethereum chain ID
+    #[clap(long)]
+    chain_id: u64,
+
+    /// Ethereum Wallet Private Key
+    #[clap(long, env)]
+    eth_wallet_private_key: String,
+
+    /// Node RPC URL
+    #[clap(long)]
+    rpc_url: String,
+
+    /// Application's contract address on Ethereum
+    #[clap(long)]
+    contract: String,
+
+    /// The proposal ID (32 bytes, hex-encoded)
+    #[clap(long)]
+    proposal_id: String,
+
+    /// The votes data (hex-encoded, multiple of 100 bytes)
+    #[clap(long)]
+    votes_data: String,
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+    // Parse CLI Arguments: The application starts by parsing command-line arguments provided by the user.
+    let args = Args::parse();
+
+    // Create a new transaction sender using the parsed arguments.
+    let tx_sender = TxSender::new(
+        args.chain_id,
+        &args.rpc_url,
+        &args.eth_wallet_private_key,
+        &args.contract,
+    )?;
+
+    // Decode the hex-encoded proposal ID and votes data
+    let proposal_id = decode(&args.proposal_id).context("Failed to decode proposal ID")?;
+    let votes_data = decode(&args.votes_data).context("Failed to decode votes data")?;
```

Comment:
In cases like this, I like to use the fully qualified name. It saves the effort of having to look for which `decode` method this is.
```suggestion
    let proposal_id = hex::decode(&args.proposal_id).context("Failed to decode proposal ID")?;
    let votes_data = hex::decode(&args.votes_data).context("Failed to decode votes data")?;
```